zerver/worker/queue_processors.py

```diff
@@ -115,25 +115,77 @@ def on_failure(event: Dict[str, Any]) -> None:
 
 class QueueProcessingWorker(ABC):
     queue_name = None  # type: str
+    CONSUME_ITERATIONS_BEFORE_UPDATE_STATS_NUM = 2
 
     def __init__(self) -> None:
         self.q = None  # type: SimpleQueueClient
         if self.queue_name is None:
             raise WorkerDeclarationException("Queue worker declared without queue_name")
+        self.queue_last_emptied_timestamp = time.time()
+        self.consumed_since_last_emptied = 0
+        self.recent_consume_times = deque(maxlen=50)  # type: MutableSequence[Tuple[int, float]]
+        self.consume_interation_counter = 0
 
     @abstractmethod
     def consume(self, data: Dict[str, Any]) -> None:
         pass
 
-    def consume_wrapper(self, data: Dict[str, Any]) -> None:
+    def do_consume(self, consume_func: Callable[[List[Dict[str, Any]]], None],
+                   events: List[Dict[str, Any]]) -> None:
         try:
-            self.consume(data)
+            time_start = time.time()
+            consume_func(events)
```

Comment:
I guess the idea of having this `consume_func` is it lets us both share code with `LoopQueueProcessingWorker` and also lets us do automated tests for this logic by passing in a no-op `consume_func` function?  If so, makes sense to me.

Reply:
Code sharing was the idea, the no-op approach for tests is a great idea that I haven't thought of :slightly_smiling_face: